backend/cmd/headlamp.go

```diff
@@ -1384,3 +1388,122 @@ func absPath(path string) (string, error) {
 
 	return filepath.Join(currentUser.HomeDir, path[2:]), nil
 }
+
+//nolint:funlen
+func (c *HeadlampConfig) handleNodeDrain(w http.ResponseWriter, r *http.Request) {
+	var drainPayload struct {
+		Cluster  string `json:"cluster"`
+		NodeName string `json:"nodeName"`
+	}
+	// get nodeName and cluster from request body
+	err := json.NewDecoder(r.Body).Decode(&drainPayload)
+	if err != nil {
+		http.Error(w, "Error decoding payload", http.StatusBadRequest)
+		return
+	}
+
+	if drainPayload.NodeName == "" {
+		http.Error(w, "nodeName is required", http.StatusBadRequest)
+		return
+	}
+
+	if drainPayload.Cluster == "" {
+		http.Error(w, "clusterName is required", http.StatusBadRequest)
+		return
+	}
+	// get token from header
+	token := r.Header.Get("Authorization")
+
+	ctxtProxy := c.contextProxies[drainPayload.Cluster]
+
+	clientset, err := ctxtProxy.context.getClientSetToInteractWithKubernetesAPIServer(token)
+	if err != nil {
+		http.Error(w, "Error getting client", http.StatusInternalServerError)
+		return
+	}
+
+	_, err = w.Write([]byte("Drain node request accepted"))
+
+	if err != nil {
+		http.Error(w, "Error writing response", http.StatusInternalServerError)
+		return
+	}
+
+	go func() {
+		nodeClient := clientset.CoreV1().Nodes()
+		ctx := context.Background()
+		cacheKey := uuid.NewSHA1(uuid.Nil, []byte(drainPayload.NodeName+drainPayload.Cluster)).String()
```

Comment:
The cacheKey has to be generated in the request response cycle and be sent to the client.